### DISCUSSION TITLE
SQL formatter for VScode

### DISCUSSION THREAD

**Discussion by ManUtopiK** [hasura/graphql-engine#8563] [General]
Hi!

I try to find an VScode extension to format my sql file with the same format used by Hasura console.
There is many dialect to format with: (sql: Standard SQL, n1ql: Couchbase N1QL, db2: IBM DB2, pl/sql: Oracle PL/SQL)

Which dialect Hasura console use?
Do you use an extension to format sql code? Which one?

Thanks.

**Comment by soorajshankar** [comment]
Hey @ManUtopiK, 

I could think of 2 possibilities here, 
1. Hasura console has a `Raw SQL` page (as shown below)where you can execute queries, 
  ![image](https://user-images.githubusercontent.com/8408875/173746627-a154d67b-9c41-425f-9982-f189ac709d35.png)
  (this UI only has syntax highlighting functionality, and automatic code formatting is not done here) 

2. If the console is running in CLI mode we store the SQL migrations in the file system. 

Could you clarify which one did you mean? 

Thanks

**Reply by ManUtopiK** [reply]
When we made changes in Hasura console in CLI mode, migration is written in the file system.
But I often edit those files to merge them, add some fix or improvement.
I would like to format on save with the same code formatting done by Hasura cli.
So I was wondering what is the dialect used by Hasura to format those files.
